set/name_trie.go

```diff
@@ -0,0 +1,96 @@
+package set
+
+import (
+    "sort"
+)
+
+type NameTrie struct {
+    children   map[rune]*NameTrie
+    terminates bool
+}
+
+func createNameTrie() *NameTrie {
+    return &NameTrie{make(map[rune]*NameTrie), false}
+}
+
+func (tree *NameTrie) Exists(name string) bool {
+    nameSlice := []rune(name)
+    node, ok := tree.traverse(nameSlice, false)
+    return ok && node.terminates
+}
+
+func (tree *NameTrie) Insert(name string) {
+    nameSlice := []rune(name)
+    node, ok := tree.traverse(nameSlice, true)
+    if ok {
+        node.terminates = true
+    }
+}
+
+func (tree *NameTrie) Delete(name string) {
+    nameSlice := []rune(name)
+    node, ok := tree.traverse(nameSlice, false)
+    if ok {
+        node.terminates = false
+    }
+}
+
+func (node *NameTrie) OrderedChildren() []rune {
+    keys := []string{}
+    for suffix := range(node.children){ keys = append(keys, string(suffix)) }
```

Comment:
👍 

I found a similar sample online when I was searching for a good way to do this; I wasn't sure how it ought to look.

Is the code here idiomatic itself, other than the style issue?